File: Udemy.Core/Entities/Ask.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Udemy.Core.Entities;
public class Ask : BaseEntity
{
    [StringLength(20)]
    public string Title { get; set; }
    public string Content { get; set; }

    public int CourseId { get; set; }
    [ForeignKey("CourseId")]
    public Course Course { get; set; }

    public int UserId { get; set; }
    [ForeignKey("UserId")]
    public ApplicationUser User { get; set; }

    public List<Answer> Answers { get; set; }
}

File: Udemy.Client/src/app/lib/models/ask.ts
export interface Ask {
  id: number;
  userId: number;
  userName: string;
  title: string;
  content?: string;
  createdDate?: string;
  answers: Answer[]
}

export interface Answer{
  id: number;
  askId: number;
  userId: number;
  userName: string;
  content: string;
  createdDate?: string;
}

export interface CreateAsk {
  title: string;
  content: string;
}

export interface UpdateAsk {
  id: number;
  title: string;
  content: string;
}

export interface CreateAnswerModel {
  content: string;
}
